ui/screens.py

```diff
@@ -1391,6 +1403,8 @@ def registerPool(wallet):
                 app.appWindow.showSuccess("pool authorized")
                 wallet.openAccount.setPool(pool)
                 wallet.save()
+                # Notify that vote data should be updated.
+                self.app.emitSignal(ui.PURCHASEINFO_SIGNAL)
```

Comment:
This is actually a great solution. You may know this, but I'll say it anyway. The Qt UI cannot be updated from outside of the Qt main thread. One advantage of using the signal system, is that all registered callbacks (or "slots" if you prefer) will be run in the main loop, so you can always update UI elements from signal callbacks. 

To note here, you are also updating the UI when you call `showSuccess`, but that function is also written as a slot in Qt's slot/signal system, so again, safe to use from another thread such as here.

Reply:
I did not know.  Thank you.  This information will save me time in the future.